reana_server/scheduler.py

```diff
@@ -12,17 +12,70 @@
 import logging
 
 from bravado.exception import HTTPBadGateway, HTTPNotFound
+from kubernetes.client.rest import ApiException
+from sqlalchemy.exc import SQLAlchemyError
+
+from reana_commons.config import REANA_MAX_CONCURRENT_BATCH_WORKFLOWS
 from reana_commons.consumer import BaseConsumer
-from reana_commons.tasks import reana_ready
-from reana_db.database import Session
-from reana_db.models import Workflow
+from reana_commons.k8s.api_client import current_k8s_corev1_api_client
+from reana_db.models import Workflow, WorkflowStatus
 
 from reana_server.api_client import (
     current_rwc_api_client,
     current_workflow_submission_publisher,
 )
 
 
+def check_predefined_conditions():
+    """Check Kubernetes predefined conditions for the nodes."""
+    try:
+        node_info = json.loads(
+            current_k8s_corev1_api_client.list_node(
+                _preload_content=False
+            ).data.decode()
+        )
+        for node in node_info["items"]:
+            # check based on the predefined conditions about the
+            # node status: MemoryPressure, OutOfDisk, KubeletReady
+            #              DiskPressure, PIDPressure,
+            for condition in node.get("status", {}).get("conditions", {}):
+                if not condition.get("status"):
+                    return False
+    except ApiException as e:
+        logging.error("Something went wrong while getting node information.")
+        logging.error(e)
+        return False
+    return True
+
+
+def check_running_reana_workflows_count():
```

Comment:
Minor: By reading the name of this function is not clear what it returns for me. Would something like `exceedes_max_reana_workflow_count` be more understandable?

Reply:
Since the logic is a bit convoluted, it would have to be `doesnt_exceed_max_reana_workflow_count` (amended). Once we tackle https://github.com/reanahub/reana-server/pull/341#discussion_r590541272 we can do some clean up of these condition checks.